libraries/botbuilder-dialogs-adaptive/src/converters/dialogListConverter.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Converter, Dialog } from 'botbuilder-dialogs';
+import { ResourceExplorer } from 'botbuilder-dialogs-declarative';
+
+export class DialogListConverter implements Converter<string[], Dialog[]> {
+    public constructor(private readonly _resourceExplorer: ResourceExplorer) {}
+
+    public convert(value: string[] | Dialog[]): Dialog[] {
+        const results: Dialog[] = [];
+
+        value.forEach((item: string | Dialog) => {
```

Comment:
Use `.map` when constructing an array of types.

```typescript
return value.map(item => {
  if (item instanceof Dialog) {
    return item;
  }
  ...
})
```

Reply:
There is an error when using `.map`
This expression is not callable.
  Each member of the union type '(<U>(callbackfn: (value: string, index: number, array: string[]) => U, thisArg?: any) => U[]) | (<U>(callbackfn: (value: Dialog<{}>, index: number, array: Dialog<{}>[]) => U, thisArg?: any) => U[])' has signatures, but none of those signatures are compatible with each other.ts(2349)

Reply:
You might need to add the same callback signature above `value.map((item: string | Dialog) => { ... })`

Reply:
I think the problem is: the input type of convert is `string[] | Dialog[]` not `(string | Dialog)[]`, so the child type can only be `string` or `Dialog` in a map iteration, not both. However, If I changed the input type to `(string | Dialog)[]`, it may violate the type defined in `Converter`, which is `convert(value: From | To): To`, any ideas?